proxy/pop3.go

```diff
@@ -0,0 +1,99 @@
+// pop3.go - POP3 + mixnet proxy server.
+// Copyright (C) 2017  David Stainton.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package proxy provides mixnet client proxies
+package proxy
+
+import (
+	"net"
+	"strings"
+
+	"github.com/katzenpost/client/pop3"
+	"github.com/katzenpost/client/storage"
+)
+
+// Pop3BackendSession is our boltdb backed implementation
+// of our pop3 BackendSession interface
+type Pop3BackendSession struct {
```

Comment:
Nothing in this file actually feels specific to pop3. I wonder if this can be just a 'mailacct' more generally, that you plug the pop3.NewSession into?

Reply:
this is the small section of code that uses our pop3 library that yawning wrote. each of these structs is implementing a specific interface require by his pop3 library. I could certainly change the names of these struct types if you have some suggestions. Otherwise I'm thinking to leave it as is.